### PR TITLE
fix(ui): Fix undefined project on releases onboarding

### PR DESCRIPTION
**Pull request by matejminar**
Fixes JAVASCRIPT-23CA
Fixes JAVASCRIPT-23CB

### CODE REVIEW

File: src/sentry/static/sentry/app/views/releases/list/releaseLanding.tsx [hunk 1]
```diff
@@ -122,7 +122,7 @@ class Promo extends React.Component<PromoProps> {
       eventKey: 'releases.landing_card_viewed',
       eventName: 'Releases: Landing Card Viewed',
       organization_id: parseInt(organization.id, 10),
-      project_id: parseInt(project.id, 10),
+      project_id: project && parseInt(project.id, 10),
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/releases/list/releaseLanding.tsx:125]: Could we change the type to be optional and callout that the HoC is wrong in this case.

Comment (matejminar) [reply src/sentry/static/sentry/app/views/releases/list/releaseLanding.tsx:125]: I can do that in a followup PR, this is a hotfix.

Comment (markstory) [reply src/sentry/static/sentry/app/views/releases/list/releaseLanding.tsx:125]: I think this is good for a short term fix. Updating the types of withProject and all its consumers is a good follow up.

**Review by markstory — APPROVED** [review]

**Review by evanpurkhiser — APPROVED** [review]

File: src/sentry/static/sentry/app/views/releases/list/releaseLanding.tsx (hunk 1 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/views/releases/list/releaseLanding.tsx:125]: yep

**Pull request merged**